src/vmm/src/device_manager/persist.rs

```diff
@@ -429,13 +433,13 @@ mod tests {
             };
             insert_vsock_device(&mut vmm, &mut cmdline, &mut event_manager, vsock_config);
             // This will be used by the restored device and will cleanup the UDS when test ends.
-            _tmp_sock_file = orig_tmp_sock_file.clone();
+            _tmp_sock_file = orig_tmp_sock_file;
 
             vmm.mmio_device_manager
                 .save()
                 .serialize(&mut buf.as_mut_slice(), &version_map, 1)
                 .unwrap();
-            vmm.mmio_device_manager.clone()
+            vmm.mmio_device_manager
```

Comment:
I think this is the root cause of the problem with this test 😄 
The `clone()` call actually does something extra, as you can see above: it only clones the necessary parts of the `MmioDeviceManager`.

You could add something like this above the line and it should be fine:
```
#[allow(clippy::redundant_clone)]
/// We only want to keep the device map from the original MmioDeviceManager.
```

Reply:
Good find. I'll revert the clone here to see how it affects the CI test.

Reply:
Hmmm, if it isn't actually doing cloning, it shouldn't be implemented as `Clone`. We can instead have a function implemented on the `DeviceManager` called `soft_clone` or something like that. This way we don't confuse clippy either. What do you think?